cmd/shutdown.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"context"
+	"io"
+	"time"
+)
+
+//GracefulShutdown implements releasing all resouces it got from system, finish all request handling and return responses when service stopping.
+func gracefulShutdown(timeout time.Duration, components []io.Closer) error {
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+
+	for _, component := range components {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			cancel()
+			return err
```

Comment:
return ctx.Err()

Reply:
oh, okay, I wrote so because of linter, but I think your varient better too